src/jit/morph.cpp

```diff
@@ -4766,15 +4766,33 @@ GenTreePtr Compiler::fgMorphMultiregStructArg(GenTreePtr arg, fgArgTabEntryPtr f
 #endif
 
 #ifdef _TARGET_ARM_
-    if (fgEntryPtr->isSplit)
+    if ((fgEntryPtr->isSplit && fgEntryPtr->numSlots + fgEntryPtr->numRegs > 4) ||
+        (!fgEntryPtr->isSplit && !fgEntryPtr->isHfaRegArg && fgEntryPtr->regNum == REG_STK))
```

Comment:
`!fgEntryPtr->isHfaRegArg` is needless, because `fgEntryPtr->isHfaRegArg` is always false when `fgEntryPtr->regNum == REG_STK`.